src/main/resources/alfresco/ibatis/org.alfresco.repo.domain.dialect.Dialect/metadata-query-SqlMap.xml

```diff
@@ -1,11 +1,11 @@
-<?xml version="1.0" encoding="UTF-8" ?>
-
-<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
-
-<mapper namespace="alfresco.metadata.query">
-
-  <select id="select_byDynamicQuery" fetchSize="200" parameterType="org.alfresco.repo.search.impl.querymodel.impl.db.DBQuery" resultMap="alfresco.node.result_NodeRef">
-      <include refid="sql_select_byDynamicQuery"/>   
-  </select>
-
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+
+<mapper namespace="alfresco.metadata.query">
+
+  <select id="select_byDynamicQuery" fetchSize="200" parameterType="org.alfresco.repo.search.impl.querymodel.impl.db.DBQuery" resultMap="alfresco.node.result_NodeRef">
+      <include refid="sql_select_byDynamicQuery"/>   LIMIT #{queryLimit} OFFSET #{skipCount}

```

Comment:
Is the LIMIT clause ANSI standard SQL? Will this work for Oracle?
https://www.postgresql.org/docs/current/sql-select.html#SQL-LIMIT

Reply:
Good observation, that syntax won't work for Oracle. I didn't think about that.

Reply:
This will work with Oracle now as I provided the query template for its dialect

https://github.com/Alfresco/alfresco-repository/blob/2d01a54daf279927dfc05450e10b5f7349caab97/src/main/resources/alfresco/ibatis/org.alfresco.repo.domain.dialect.Oracle9Dialect/metadata-query-limit-clause-SqlMap.xml#L1-L18

I also ensured that the related tests in [DBQueryTest](https://github.com/Alfresco/alfresco-repository/blob/feature/REPO-5335_Pagination_Optimisation/src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryTest.java) were working after my changes on all supported databases.